src/inmanta/deploy/scheduler.py

```diff
@@ -677,7 +685,16 @@ async def report_resource_state(
                 self._deploying_latest.remove(resource)
                 state.deployment_result = deployment_result
                 self._work.finished_deploy(resource)
-                if deployment_result is DeploymentResult.DEPLOYED:
+                # Check if we need to mark a resource as transiently blocked
+                # Or if we can unblock a transiently blocked resource (its dependencies now succeed)
+                if skipped_for_dependency and state.blocked is not BlockedStatus.YES:
+                    dependencies = await self._get_last_non_deploying_state_for_dependencies(resource=resource)
+                    state.blocked = (
+                        BlockedStatus.NO
+                        if all(state is const.ResourceState.deployed for state in dependencies.values())
+                        else BlockedStatus.TRANSIENT
+                    )
+                if deployment_result is DeploymentResult.DEPLOYED or state.blocked is not BlockedStatus.NO:
```

Comment:
and here is where we assign the `TRANSIENT` state when applicable and discard anything that is not unblocked